janitor/io.py

```diff
@@ -689,3 +691,187 @@ def _object_to_dict(obj):
             data[key] = _object_to_dict(value)
         return data
     return obj
+
+
+#################################################################
+
+
+def read_archive(
+    file_path: str,
+    extract_to_df: bool = True,
+    file_type: str | None = None,
+    selected_files: list[str] | None = None,
+) -> pd.DataFrame | list[str]:
```

Comment:
i think we should allow for more flexibility, via kwargs, where you can pass extra info to `read_csv`, `read_excel`, `read_parquet`, etc